test/Datadog.Trace.Tests/TracerSettingsTests.cs

```diff
@@ -74,5 +70,37 @@ public void DDVarTakesPrecedenceOverDDTags(string envKey, string tagKey)
 
             Assert.Equal(span.GetTag(tagKey), envValue);
         }
+
+        [Theory]
+        [InlineData("", true)]
+        [InlineData("1", true)]
+        [InlineData("0", false)]
+        public void TraceEnabled(string value, bool areTracesEnabled)
+        {
+            var originalValue = Environment.GetEnvironmentVariable(ConfigurationKeys.TraceEnabled);
+
+            try
+            {
+                Environment.SetEnvironmentVariable(ConfigurationKeys.TraceEnabled, value);
+
+                var settings = new TracerSettings(new EnvironmentConfigurationSource());
+
+                Assert.Equal(areTracesEnabled, settings.TraceEnabled);
```

Comment:
Should we be concerned about setting env vars in unit tests? Could it cause issues with multiple tests running in parallel? The alternative would be to use another `IConfigurationSource` like `NameValueConfigurationSource`.

Reply:
Yeah it's probably much safer to use `NameValueConfigurationSource`

Reply:
My current PR just got bit by using `EnvironmentConfigurationSource` because multiple xUnit test collections (test classes) were running in parallel and setting the same environment variables to different values. See 6022c4a7bcae75335447641d01ef5d67810dc32e for how I modified them

Reply:
Oh I didn't consider that multiple tests could run in parallel in the same process. You're right, gonna change that